src/basic/proc-cmdline.c

```diff
@@ -39,6 +39,39 @@ int proc_cmdline(char **ret) {
                 return read_one_line_file("/proc/cmdline", ret);
 }
 
+
+/* note; similar function is also in libshared, but libbasic cant't depend
+* on libshared */
+static bool is_efi_secure_boot() {
+        _cleanup_free_ void *v = NULL;
+        uint8_t b;
+        size_t s;
+        int r;
+
+        r = efi_get_variable(EFI_VENDOR_GLOBAL, "SecureBoot", NULL, &v, &s);
+        if (r < 0)
+                return false;
+
+        if (s != 1)
+                return false;
+
+        b = *(uint8_t *)v;
+        return !!b > 0;
+}
```

Comment:
please just move is_efi_secure_boot() and the related calls from shared/efi-loader.c to basi/efivars.c, i.e. building on 0bb2f0f10e627d33cf3872b2ec35c0454a517ca1